app/main/views/g_cloud.py

```diff
@@ -265,3 +276,115 @@ def search_services():
+    clean_request_query_params = clean_request_args(request.args, filters.values(), lots_by_slug)
+
+    # Retrieve results so we can display SearchSummary
+    search_api_response = search_api_client.search_services(
+        index=framework['slug'],
+        **build_search_query(request.args, filters.values(), content_manifest, lots_by_slug)
+    )
+
+    search_summary = SearchSummary(search_api_response['meta']['total'], clean_request_query_params.copy(),
+                                   filters.values(), lots_by_slug)
+
+    # Embed search api URL in form so that the next view can be aware of it
+    # This is maybe not ideal
+    search_api_url = search_api_client.get_search_url(
+        index=framework['slug'],
+        **build_search_query(request.args, filters.values(), content_manifest, lots_by_slug)
+    )
+
+    return render_template(
+        'direct-award/save-search.html',
+        search_summary=search_summary,
+        search_api_url=search_api_url,
+    )
+
+
+@direct_award.route('/projects/create', methods=['POST'])
+def project_create():
+    try:
+        api_project = data_api_client.create_direct_award_project(user_id=current_user.id,
+                                                                  user_email=current_user.email_address,
+                                                                  project_name=request.form.get('project_name',
+                                                                                                'My new project'))
+
+    except HTTPError as e:
+        abort(e.status_code)
+
+    project = api_project['project']
+
+    try:
+        data_api_client.create_direct_award_project_search(user_id=current_user.id,
+                                                           user_email=current_user.email_address,
+                                                           project_id=project['id'],
+                                                           search_url=request.form['search_api_url'])
+    except HTTPError as e:
+        abort(e.status_code)
+
+    flash(PROJECT_CREATED_MESSAGE, 'success')
+
+    return redirect(url_for('.view_project',
+                            project_id=project['id']
+                            ))
+
+
+@direct_award.route('/projects/<int:project_id>', methods=['GET'])
+def view_project(project_id):
+    # Get core data
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(all_frameworks, 'g-cloud')
+    content_manifest = content_loader.get_manifest(framework['slug'], 'search_filters')
+    lots_by_slug = framework_helpers.get_lots_by_slug(framework)
+
+    # Get the requested Direct Award Project.
+    project = data_api_client.get_direct_award_project(user_id=current_user.id, project_id=project_id)['project']
+    if not is_direct_award_project_accessible(project, current_user.id):
```

Comment:
The API currently doesn't enforce that check for this endpoint, and apparently access control is done frontend-side, so it's done here to be consistent with briefs.

Reply:
It looked to me like the API will - as long as the client passes the user id in it will filter: https://github.com/alphagov/digitalmarketplace-api/blob/master/app/main/views/direct_award.py#L23

But having checks in the frontend definitely doesn't hurt.

Reply:
This isn't calling list_projects, it's calling get_project (by id).

Reply:
Ah righto, sorry. My bad.

Reply:
In that case it looks like the API client is wrong - this method shouldn't take or pass on a user ID: https://github.com/alphagov/digitalmarketplace-apiclient/blob/master/dmapiclient/data.py#L795

Reply:
Because the API will just completely ignore it: https://github.com/alphagov/digitalmarketplace-apiclient/blob/master/dmapiclient/data.py#L795

Reply:
Yeah, I will fix that up shortly (but it's a lower priority at the moment).